assets/js/admin/course-video-progression-sidebar.js

```diff
@@ -0,0 +1,73 @@
+/**
+ * WordPress dependencies
+ */
+import { PluginDocumentSettingPanel } from '@wordpress/edit-post';
+import { __ } from '@wordpress/i18n';
+import { SelectControl, ToggleControl } from '@wordpress/components';
+import { useEntityProp } from '@wordpress/core-data';
+
+/**
+ * A hook that provides a value from course meta and a setter for that value.
+ *
+ * @param {string} metaName The name of the meta.
+ *
+ * @return {Array} An array containing the value and the setter.
+ */
+const useCourseMeta = ( metaName ) => {
```

Comment:
I took this helper function from course-theme. Looks like a good candidate to be extracted.
But I'm not sure where should I put it. Maybe `assets/shared/helpers`?

Reply:
Hmmm...not sure. @renatho any opinion on this?

Reply:
Hey there!

Good idea! It would be interesting to extract this to use the same function in both places. There's a folder with some react hooks, probably it would be a good fit for that: `assets/react-hooks/`.